qutip/steadystate.py

```diff
@@ -72,6 +73,11 @@
     from qutip._mkl.spsolve import (mkl_splu, mkl_spsolve)
 
 
+_GMRES_CALLBACK_TYPE = {}
```

Comment:
Nitpicking: Could we make `_GMRES_CALLBACK_TYPE = None` and then set `= "legacy"` if needed and still construct the `extra` dictionary in the function that uses it? It just feels a bit odd to have a mutable global variable with a slightly misleading name (i.e. it's not just the callback type -- it's a dict of extra options).

Reply:
I actually renamed it because I thought it was clearer what it was this way round - it wasn't something a user could pass in, so it actually only _did_ have one possible kwarg in it. When I first read the file I assumed that it could be influenced by a user.

Agree that the global state isn't ideal - I could instead redefine `gmres` at the top to wrap the scipy implementation and do the version check there?